src/ray/raylet/wait_manager.h

```diff
@@ -0,0 +1,103 @@
+// Copyright 2022 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "ray/common/asio/instrumented_io_context.h"
+#include "ray/common/id.h"
+
+namespace ray {
+namespace raylet {
+
+/// This class is responsible for handling wait requests (e.g. `ray.wait`).
+/// It is not thread safe and is expected to run ONLY in
+/// the NodeManager.io_service_ thread.
+class WaitManager {
+ public:
+  WaitManager(instrumented_io_context *io_service,
```

Comment:
It' nicer for testing to pass a lambda for creating a timer. You can look at PullManager for an example.